app/views/tag/_subscribe_button.html.erb

```diff
@@ -1,19 +1,13 @@
 <!-- This button is used to display tags in a popover and subscribe to multiple tags using a single button -->
-<% if current_user %>
-  <a id="popover_button" class="btn btn-outline-secondary btn-sm sub" >Subscribe 
-    <i class="fa fa-plus"></i> 
-  </a>
+<a id="popover_button" class="btn btn-default btn-xs sub requireLogin" > Subscribe <i class="fa fa-plus"></i> </a>
```

Comment:
If the user is not logged in, the modal is triggered here.

Reply:
OK, so, this is what I'd like to modify - if there's no logged-in user, we should still wait until after the popover is open to prompt a login modal, and we should have all the checkboxes checked (we can later modify this so it's only the top 3, maybe? But... later!)

That way, if someone comes to the site for the first time, clicks Subscribe, they can in as few steps as possible, get registered AND subscribed to the most relevant tags on their starting page, rather than having to register and then come back to subscribe. Does this make sense? 

So, a non-logged-in user would still see:

![image](https://user-images.githubusercontent.com/24359/57144658-89bb1180-6d8f-11e9-8e97-9c586e9253a9.png)


And they'd be prompted to log in only when pressing the blue `Subscribe` button. This would establish a `return_to=/subscribe/multiple/tag/arduino,games`, in the login/signup modal. Make sense?

Reply:
Oh okay, I think I get you, however it might be a little problematic since `requireLogin` was not working on this, but lemme take a stab at it again and get back to you! Thanks for explaining so clearly!

Reply:
Ah yes - so, I think the reason it wasn't working is that the popover is constructed /after/ this code is run, which is the code which makes `requireLogin` work:

https://github.com/publiclab/plots2/blob/e190eae1ce7bf215b99b6efe7f828e17deb3213e/app/views/user_sessions/_form.html.erb#L74-L81

So, we have to do something similar, re-running it, or re-running it with a narrower scope just for the popover, you know?

Reply:
Yes, I think I get you somewhat, but I'll have to try it hands on to exactly get it!

Reply:
Hey, I am facing a bit of a problem in understanding the flow here, could you please help me out @jywarren, to identify what is going on and what should be done so that the `requireLogin`class activates the modal after the popover is rendered. Should we add the class dynamically? Or should we build it from scratch for this case?